### PR TITLE
Create Authentication providers for API ML

### PR DESCRIPTION
**Pull request by janan07**
Describe your pull request here: This PR creates the article authentication-providers-for-apiml.md as a stand-alone article

List the file(s) included in this PR: authentication-providers-for-apiml.md, authentication-for-apiml-services.md, sidebars.js

After creating the PR, follow the instructions in the comments.

### CODE REVIEW

File: docs/user-guide/authentication-providers-for-apiml.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+* When z/OSMF is not available, we recommend you use the SAF Authentication provider. With the SAF provider, the API Gateway acts as the authentication service. The provided credentials are validated directly by API Gateway via SAF APIs.
+:::
+
+## z/OSMF Authentication Provider
+
+The `z/OSMF Authentication Provider` allows the API Gateway to authenticate with the z/OSMF service. The user needs z/OSMF access in order to authenticate.
+
+Use the following properties of the API Gateway to enable the `z/OSMF Authentication Provider`:
+```
+apiml.security.auth.provider: zosmf
```
Comment (balhar-jakub) [inline docs/user-guide/authentication-providers-for-apiml.md:27]: This needs to be updated to:
components.gateway.security.auth.provider: zosmf
components.gateway.security.auth.zosmfServiceId: zosmf

File: docs/user-guide/authentication-providers-for-apiml.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+apiml.security.auth.zosmfServiceId: zosmf  # Replace me with the correct z/OSMF service id
+```
+## SAF Authentication Provider
+
+The `SAF Authentication Provider` allows the API Gateway to authenticate directly with the z/OS SAF provider that is installed on the system. The user needs a SAF account to authenticate. 
+
+Use the following property of the API Gateway to enable the `SAF Authentication Provider`:
+```
+apiml.security.auth.provider: saf
```
Comment (balhar-jakub) [inline docs/user-guide/authentication-providers-for-apiml.md:36]: components.gateway.security.auth.provider: saf

File: docs/user-guide/authentication-providers-for-apiml.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+```
+**Note:** To provide your own implementation of the SAF IDT provider, see the [Implement new SAF provider](implement-new-saf-provider.md) guidelines.
```
Comment (balhar-jakub) [inline docs/user-guide/authentication-providers-for-apiml.md:38]: Remove this note. This is unrelated. The SAF IDT provider is used for something else.

File: docs/user-guide/authentication-providers-for-apiml.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+:::info Required roles: system administrator, security administrator
+:::
+
+Choose from the following providers to handle authentication for the API Gateway:  
+* [z/OSMF Authentication Provider](#z-osmf-authentication-provider)
+* [SAF Authentication Provider](#saf-authentication-provider)
+
+:::note
+For development purposes, a dummy authentication provider is also available. This provider is not intend for production purposes.
```
Comment (balhar-jakub) [inline docs/user-guide/authentication-providers-for-apiml.md:11]: The initial version of the file is here: https://github.com/zowe/docs-site/blob/janan07-create-auth-providers-for-apiml-section/docs/extend/extend-apiml/api-mediation-layer-development-setup.md

File: docs/user-guide/authentication-providers-for-apiml.md (hunk 1 above)
Comment (janan07) [reply docs/user-guide/authentication-providers-for-apiml.md:27]: change applied

**Review by balhar-jakub — APPROVED** [review]
